### PR TITLE
init Limesuite, SoapySDR

### PR DESCRIPTION
**Pull request by roosemberth**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by ryantm** [comment]
@GrahamcOfBorg build soapysdr limesuite

File: pkgs/applications/misc/limesuite/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  name = "LimeSuite-${version}";
+
+  src = fetchurl {
+    url = "https://github.com/myriadrf/LimeSuite/archive/v${version}.tar.gz";
+    sha256 = "0fiadrkz5rh1yngjw6q6mg55k375pvp4ba9fkj3x8h3wh9ym32i6";
+  };
+
+  buildInputs = [ libusb cmake wxGTK30 mesa libX11 soapysdr sqlite doxygen gnuplot ];
+
+  configurePhase = ''
```
Comment (markuskowa) [inline pkgs/applications/misc/limesuite/default.nix]: Defining a configure phase is not necessary here. You can use `cmakeFlags = [ "-DENABLE_STREAM_UNITE=ON" ]` instead.

File: pkgs/applications/misc/limesuite/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchurl, libusb, cmake, wxGTK30, mesa, libX11, soapysdr, sqlite, doxygen, gnuplot ? null }:
+
+stdenv.mkDerivation rec {
+  version = "17.12.0";
+  name = "LimeSuite-${version}";
+
+  src = fetchurl {
```
Comment (markuskowa) [inline pkgs/applications/misc/limesuite/default.nix]: I think `fetchFromGitHub` is the preferred method:
  src = fetchFromGitHub {
    owner = "myriadrf";
    repo = "LimeSuite";
    rev = "v${version}";
    sha256 = "1rd79xxnwr62fmy2vq895jyiqnz6lm4p8bl09adjqqm4y5npryf9";
  };

File: pkgs/applications/misc/soapysdr/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  buildInputs = [ cmake python2 python3 python2Packages.numpy python3Packages.numpy swig ncurses doxygen];
+
+  cmakeFlags = [ "-DCMAKE_BUILD_TYPE=Release"
+                 "-DUSE_PYTHON_CONFIG=ON"
+                 "-DPYTHON_EXECUTABLE=${python2}/bin/python"
+                 "-DPYTHON3_EXECUTABLE=${python3}/bin/python"
+               ];
+
+  configurePhase = ''
```
Comment (markuskowa) [inline pkgs/applications/misc/soapysdr/default.nix]: Same as above. Defining `cmakeFlags` should be enough.

**Comment by markuskowa** [comment]
How do you keep track of the SoapySDR plugins? Every application that wants to use the soapysdr backend needs to have (i) an environment variable set pointing to all plugin derivations or (ii) have all plugins a the soapysdr compiled in path.

**Pull request closed without merging**